module/zstd/zstd.c

```diff
@@ -399,18 +403,20 @@ zstd_decompress_level(void *s_start, void *d_start, size_t s_len, size_t d_len,
 	enum zio_zstd_levels zstdlevel;
 	int16_t levelcookie;
 	uint32_t version;
+	uint32_t bufsize;
+	const struct zstd_header *hdr = (const struct zstd_header *)s_start;
 
-	const char *src = s_start;
-	uint32_t bufsize = BE_IN32(src);
+	/* Read buffer size */
+	bufsize = BE_32(hdr->size);
```

Comment:
I'm not sure about that change from `BE_IN32` to `BE_32`

Reply:
but i'm sure. BE_32 uses immediate integer BE_IN32 uses pointers as source

Reply:
ack